ospfd/ospf_vty.c

```diff
@@ -4717,34 +4717,45 @@ static int show_ip_ospf_neighbor_int_common(struct vty *vty, struct ospf *ospf,
 
 DEFUN (show_ip_ospf_neighbor_int,
        show_ip_ospf_neighbor_int_cmd,
-       "show ip ospf neighbor IFNAME [json]",
+       "show ip ospf [vrf <NAME>] neighbor IFNAME [json]",
        SHOW_STR
        IP_STR
        "OSPF information\n"
+       VRF_CMD_HELP_STR
        "Neighbor list\n"
        "Interface name\n"
        JSON_STR)
 {
 	struct ospf *ospf;
 	int idx_ifname = 4;
+	int idx_vrf;
 	bool uj = use_json(argc, argv);
-	struct listnode *node = NULL;
 	int ret = CMD_SUCCESS;
 	struct interface *ifp = NULL;
+	char *vrf_name = NULL;
+	vrf_id_t vrf_id = VRF_DEFAULT;
+	struct vrf *vrf = NULL;
+
+	if (argv_find(argv, argc, "vrf", &idx_vrf))
+		vrf_name = argv[idx_vrf + 1]->arg;
+	if (vrf_name == NULL || strmatch(vrf_name, VRF_DEFAULT_NAME))
+		vrf_name = NULL;
```

Comment:
I think something like this would be easier to follow: `if (vrf_name && strmatch(vrf_name, VRF_DEFAULT_NAME))`